Phase8/EFCore/SearchEngine/InvetedIndexContext.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using SearchEngine.Interfaces;
+
+namespace SearchEngine
+{
+    public class InvertedIndexContext : DbContext,IDatabaseMap<string,string>
```

Comment:
This class doesn't follow the single responsibility rule and can be divided into 2 classes. Consider a situation that you want to change the implementation of _Add, Create and ..._. This can lead to repetition of code for the DBContext segment.

Reply:
I didn't get the two responsibilities that you mentioned
IDatabaseMap has the responsibility of storing data however it wants
this class implement it with SQL
if you change the way you store data you should reimplement all methods

Reply:
Let me explain it a bit more. The two responsibilities are : _Configuring DB by inheriting DBContex  class _ and also _Implementing IDatabaseMap to customize the CRUD functionalities_. In fact, its better to extend two classes, one as a Child of DBContext and the other from IDatabaseMap, and push down related methods to each of them. Then, you can easily pass an instance of _YourContext_ to _YourMap_ class. In this way, they become completely isolated from each other.

Reply:
https://www.geeksforgeeks.org/favoring-composition-over-inheritance-in-java-with-examples/